rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -119,13 +152,60 @@ impl From<FallbackError> for ProviderError {
     }
 }
 
-fn warn_span(idx: usize, provider: &PrometheusJsonRpcClient<Http>) {
-    let _span = warn_span!("request_with_fallback", provider_index=%idx, ?provider).entered();
+async fn handle_stalled_provider(
+    fallback_provider: &HttpFallbackProvider,
+    priority: &PrioritizedProviderInner,
+) -> Result<(), <HttpFallbackProvider as JsonRpcClient>::Error> {
+    let provider = &fallback_provider.0.providers[priority.index];
+    let current_block: U64 = provider.request("eth_blockNumber", ()).await?;
```

Comment:
As an optimization to avoid always needing to make an additional eth_blockNumber call whenever any kind of RPC is made, it's probably worth also piggybacking off intentional eth_blockNumber calls that are made, & also avoiding making this eth_blockNumber RPC super frequently

e.g. something like:
* in `request`, check if the RPC being made is an `eth_blockNumber` - if it is, piggyback off this and save it in the `last_block_height`
* if it's not an `eth_blockNumber`, check when the last `eth_blockNumber` for that provider was made. If it's been X mins (maybe 2 mins or something) since we were made aware of the last `eth_blockNumber` for this provider, then make the `eth_blockNumber` request ourselves and check if it's stalled

Reply:
As discussed, I implemented the second bullet point and the first one can be a separate ticket to avoid scope creep